controllers/openshift_controller.go

```diff
@@ -1563,6 +1572,38 @@ func (eh *NodeEventHandler) Delete(ctx context.Context, event event.DeleteEvent,
 func (eh *NodeEventHandler) Generic(ctx context.Context, event event.GenericEvent, queue workqueue.RateLimitingInterface) {
 }
 
+func configMapFilterPredicate(reconciler *KataConfigOpenShiftReconciler) predicate.Predicate {
+	isRelevantConfigMap := func(obj client.Object) bool {
+		if reconciler == nil || reconciler.FeatureGates == nil {
+			return false
+		}
+
+		configMap, ok := obj.(*corev1.ConfigMap)
+		if !ok {
+			return false
+		}
+
+		relevantNamespace := configMap.Namespace == reconciler.FeatureGates.Namespace
+		relevantName := configMap.Name == reconciler.FeatureGates.ConfigMapName
```

Comment:
One thing which is unclear to me is who is setting the FeatureGates struct that is being compared here ?

Reply:
Hi @bpradipt, we are doing this at main:146 when creating the KataConfigOpenShiftReconciler.